app/preparation_to_battle/preparation.py

```diff
@@ -0,0 +1,37 @@
+class Preparation:
+    def __init__(self, knight):
+        self.knight = knight
+
+    def set_characteristics(self):
+
+        for name_knight, value in self.knight.items():
+            current_knight = {name_knight: {}}
+
+            current_knight[name_knight] = value
+
+            # apply armour
+            current_knight[name_knight]["protection"] = 0
+
+            for a in value["armour"]:
+                current_knight[name_knight]["protection"] += a["protection"]
+
+            # apply weapon
+            current_knight[name_knight]["power"] += value["weapon"]["power"]
+
+            # apply potion if exist
+            if current_knight[name_knight]["potion"] is not None:
```

Comment:
This part of function seems a little complex. Create another function to apply potion on knight